app/assets/javascripts/discourse/controllers/composer.js.es6

```diff
@@ -92,6 +111,25 @@ export default Ember.Controller.extend({
     return currentUser && currentUser.get('staff') && this.siteSettings.enable_whispers && action === Composer.REPLY;
   },
 
+  @computed("model.composeState")
+  popupMenuOptions(composeState) {
+    const self = this;
+
+    if (composeState === 'open') {
+      return _popupMenuOptionsCallbacks.map(callback => {
+        let option = callback();
+
+        if (option.condition) {
+          option.condition = self.get(option.condition);
+        } else {
+          option.condition = true;
+        }
+
+        return option;
+      });
+    }
+  },
```

Comment:
@eviltrout I updated to follow this pattern instead :) I'm not longer exposing the entire controller except for `self.get` since we need to allow plugin to add custom conditions as well.

```
Composer.reopen({
  canShow() { return true; }
})

api.addPopupMenuOptionsCallback(function() {
      return {
        action: 'someAction',
        icon: 'eye-slash',
        label: ''some label",
        condition: "canShow"
      };
});
```